CMakeLists.txt

```diff
@@ -37,14 +37,23 @@ set(CXX_STANDARD_REQUIRED ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" "${CMAKE_SOURCE_DIR}/external/cmake-modules")
 
+option(USE_STRICT_COMPILER_WARNINGS "Use a stricter compiler-warning level" OFF)
+
+if (USE_STRICT_COMPILER_WARNINGS)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wconversion -Wextra -Werror -Wno-unused-parameter -Wno-unused-function -Wno-missing-field-initializers")
+    elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wconversion -Wextra -Werror -Wno-unused-parameter -Wno-unused-function -Wno-missing-field-initializers")
+    endif ()
```

Comment:
else warning about option being unused for for other compilers

Reply:
I'll add a warning, would be cool to swap that very soon with a pendant for msvc. also gcc and clang turned out to be identical, can maybe consolidate

Reply:
done